Wikipedia/Code/NotificationsCenterViewModel.swift

```diff
@@ -89,47 +98,80 @@ final class NotificationsCenterViewModel: NSObject {
             delegate?.update(types: updateTypes)
         }
     }
+    
+    @objc private func remoteNotificationsControllerDidUpdateFilterState() {
+        //Not doing anything here yet
+        //Because filter screen disapperances call self.resetAndRefreshData from the view controller, which fetches the first page again and relays state changes back to the view controller, there's no need to react to this notification.
```

Comment:
RE: the interim load state flashing you mentioned – it seems like sometimes cases of opening the center > opening the filter settings > switching the read status from all to read and > dismissing the filters modal results in a lil' empty state loading flash.

I think if we forced the main center to reload here, while the user is likely still in the filters modal, we could prevent that. But I don't think I'm suggesting we actually do that, as it's more just masking that problem rather than directly addressing it. Probably better we just address that separately outside of this PR – what do you think?

Reply:
Yeah, I had a similar set up like that at some point, that is, updating the collection view whenever the filter state changed at all. The problem was iPad - because the modal is presented as `.pageSheet`, you could see the collection view underneath showing empty states and animating inserts/deletes with every little toggle the user did on the filter/inbox views. I found it distracting, so I updated it so that the collection view updates only when those modals are about to disappear. 

But that is a good bug you found and exactly what I was talking about, thanks for finding an instance of it. I'm not sure how to tackle it at this point, but I'll add it to my bug list and make a Phab task on it so that we can address it separately.